ujit_iface.c

```diff
@@ -506,6 +505,82 @@ ujit_disasm(VALUE self, VALUE code, VALUE from)
 }
 #endif
 
+static void
+counter_name_foreach(void (*callback)(char *, int, void *), void *data)
+{
+    char *name_reader = ujit_counter_names;
+    char *counter_name_end = ujit_counter_names + sizeof(ujit_counter_names);
+    while (name_reader < counter_name_end) {
+        if (*name_reader == ',' || *name_reader == ' ' || *name_reader == '\0') {
+            name_reader++;
+            continue;
+        }
+
+        int name_len;
+        char *name_end;
+        {
+            name_end = strchr(name_reader, ',');
+            if (name_end == NULL) name_end = strchr(name_reader, '\0');
+            if (name_end == NULL) break;
+            name_len = (int)(name_end - name_reader);
+        }
+
+        callback(name_reader, name_len, data);
+
+        name_reader = name_end;
+    }
+}
+
+
+struct collect_counters {
+    char *counter_reader;
+    char *counter_reader_end;
+    VALUE hash;
+};
+
+
```

Comment:
Definitely need at least a one-line comment explaining what each function and struct's purpose is.